src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -4671,24 +4671,30 @@ void Unit::ApplyStatPercentBuffMod(Stats stat, float val, bool apply)
 void Unit::_RegisterDynObject(DynamicObject* dynObj)
 {
     m_dynObj.push_back(dynObj);
+    if (GetTypeId() == TYPEID_UNIT && ToCreature()->IsAIEnabled)
```

Comment:
Bad habit of looking at other code and figuring as THAT code was commited, it's probably better than what I originaly wrote :P